### PR TITLE
Added localization support:

### PR DESCRIPTION
**Pull request by vicng**
- Can load translations/synonyms in TranslationManager
- Can specify translation/synonym keys in JSON block definitions
- Converted colour_blocks.json/list_blocks.json to use internationalization keys

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-ios/342)

### CODE REVIEW

**Comment by vicng** [comment]
+@corydcode +@roboerikg 

---

Review status: 0 of 17 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-ios/342#-:-KgSMY2ffkA7rlinhPfs:b4ivu6b)*

File: Source/Model/JSON/Block+JSON.swift [hunk 1]
```diff
@@ -253,11 +256,85 @@ extension Block {
...
+   Block.translatedMessage("%{non_existent_message}")  // Returns "%{non_existent_message}"
+   ```
+
+   - note: Translating a message with a key inside another key is not supported by this method
+   (eg. `"%{bky_{%bky_key2}key1}"`). It's recommended that this situation is avoided as the outcome
+   of this type of translation cannot be guaranteed.
+   - parameter message: The message to translate.
+   - returns: The translated version of `message`.
+   */
+  internal class func translatedMessage(_ message: String) -> String {
```
Comment (RoboErikG) [inline Source/Model/JSON/Block+JSON.swift]: Connotation for translation doesn't seem right, maybe 'expandedMessage'?

File: Source/Common/TranslationManager.swift [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+ manager.translation(forKey: "LISTS_INLIST")                    // Returns "in list"
+ manager.translation(forKey: "LISTS_GET_INDEX_INPUT_IN_LIST")   // Returns "in list"
+ manager.translation(forKey: "LISTS_SET_INDEX_INPUT_IN_LIST")   // Returns "in list"
+ ```
+
+ This class is designed as a singleton instance, accessible via `TranslationManager.shared`.
+ */
+@objc(BKYTranslationManager)
+public class TranslationManager: NSObject {
```
Comment (RoboErikG) [inline Source/Common/TranslationManager.swift]: I think MessageManager might be a more accurate name for this class. Translation implies it's converting between languages.

**Comment by vicng** [comment]
Review status: 0 of 17 files reviewed at latest revision, 2 unresolved discussions.

---

*[Source/Common/MessageManager.swift, line 44 at r1](https://reviewable.io:443/reviews/google/blockly-ios/342#-KgSMWBhJLx0AmbkJjXF-r1-44:-KgWF7cnmC771k6jSsjX:b-896fix) ([raw file](https://github.com/google/blockly-ios/blob/7b8292669b793d7c71bff6721b6eea94867a6e58/Source/Common/MessageManager.swift#L44)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

I think MessageManager might be a more accurate name for this class. Translation implies it's converting between languages.
</blockquote></details>

Done.

---

*[Source/Model/JSON/Block+JSON.swift, line 299 at r1](https://reviewable.io:443/reviews/google/blockly-ios/342#-KgSMWBhJLx0AmbkJjX9-r1-299:-KgWF8eyfjSCYaTMmsyv:b-896fix) ([raw file](https://github.com/google/blockly-ios/blob/7b8292669b793d7c71bff6721b6eea94867a6e58/Source/Model/JSON/Block+JSON.swift#L299)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

Connotation for translation doesn't seem right, maybe 'expandedMessage'?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-ios/342)*

**Comment by RoboErikG** [comment]
:lgtm: @corydcode can you do a sanity check?

---

Review status: 0 of 17 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-ios/342#-:-KgWIEMat-ZtxmH-dlt_:b-occqbr)*

**Comment by CoryDCode** [comment]
:lgtm:

---

Review status: 0 of 17 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-ios/342#-:-KgWIudli8EHByub0Niv:bnfp4nl)*

**Pull request merged**